### PR TITLE
Tx integrity checks refactor

### PR DESCRIPTION
**Pull request by mateigraura**
Move transaction integrity checks to TransactionHandler interface

### CODE REVIEW

**Review by sasurobert — CHANGES_REQUESTED** [review]
I think some of these changes has to be added to process/block/postporcess/intermadiateResults.go

**Review by sasurobert — DISMISSED** [review]

File: process/block/postprocess/intermediateResults.go [hunk 1]
```diff
@@ -239,11 +239,9 @@ func (irp *intermediateResultsProcessor) AddIntermediateTransactions(txs []data.
 }
 
 func (irp *intermediateResultsProcessor) checkSmartContractResultIntegrity(scr *smartContractResult.SmartContractResult) error {
-	if len(scr.RcvAddr) == 0 {
-		return process.ErrNilRcvAddr
-	}
-	if len(scr.SndAddr) == 0 {
-		return process.ErrNilSndAddr
+	err := scr.CheckIntegrity()
```
Comment (iulianpascalau) [inline process/block/postprocess/intermediateResults.go:242]: :+1: nice and clean, no more duplicated code

File: process/rewardTransaction/interceptedRewardTransaction.go [hunk 2]
```diff
@@ -96,16 +96,9 @@ func (inRTx *InterceptedRewardTransaction) processFields(rewardTxBuff []byte) er
 
 // integrity checks for not nil fields and negative value
 func (inRTx *InterceptedRewardTransaction) integrity() error {
```
Comment (iulianpascalau) [inline process/rewardTransaction/interceptedRewardTransaction.go:98]: This integrity function is no longer needed. Might call directly inRTx.rTx.CheckIntegrity() where integrity function was used

File: process/rewardTransaction/interceptedRewardTransaction_test.go [hunk 3]
```diff
@@ -313,3 +315,75 @@ func TestNewInterceptedRewardTransaction_CheckValidityShouldWork(t *testing.T) {
...
+		&mock.MarshalizerMock{},
+		&mock.HasherMock{},
+		createMockPubkeyConverter(),
+		mock.NewMultiShardsCoordinatorMock(3),
+	)
+
+	assert.Equal(t, expectedType, irt.Type())
+}
+
+func TestInterceptedRewardTransaction_String(t *testing.T) {
```
Comment (iulianpascalau) [inline process/rewardTransaction/interceptedRewardTransaction_test.go:335]: :+1:

File: process/transaction/interceptedTransaction_test.go [hunk 4]
```diff
@@ -1417,3 +1419,94 @@ func TestRelayTransaction_NotAddedToWhitelistUntilIntegrityChecked(t *testing.T)
...
+		[]byte("T"),
+		false,
+		mock.HasherMock{},
+		versioning.NewTxVersionChecker(0),
+	)
+
+	assert.Equal(t, big.NewInt(0), txin.Fee())
+}
+
+func TestInterceptedTransaction_String(t *testing.T) {
```
Comment (iulianpascalau) [inline process/transaction/interceptedTransaction_test.go:1468]: :+1:

File: process/unsigned/interceptedUnsignedTransaction.go [hunk 5]
```diff
@@ -111,20 +111,9 @@ func (inUTx *InterceptedUnsignedTransaction) processFields(uTxBuffWithSig []byte
 
 // integrity checks for not nil fields and negative value
 func (inUTx *InterceptedUnsignedTransaction) integrity() error {
```
Comment (iulianpascalau) [inline process/unsigned/interceptedUnsignedTransaction.go:113]: here too, the unexported function is no longer needed

Comment (mateigraura) [reply process/unsigned/interceptedUnsignedTransaction.go]: done

File: process/rewardTransaction/interceptedRewardTransaction.go (hunk 2 above)
Comment (mateigraura) [reply process/rewardTransaction/interceptedRewardTransaction.go]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by valentincraciun99 — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**